### DISCUSSION TITLE
Where to implement whatsapp message sending logic

### DISCUSSION THREAD

**Discussion by rodrigolfh** [django-helpdesk/django-helpdesk#1219] [Q&A] [answered]
I registered a meta app to send whatsapp messages and have an approved template and also have a working script to send a message with some hard-coded (btm) data.
Now the question is where should I implement it? I was thinking to add it to the same function that sends the emails, but where is it??

Thanks in advance!

**Comment by Benbb96** [comment, marked as answer]
I think you can use the `send` function in the `Ticket` model : https://github.com/django-helpdesk/django-helpdesk/blob/b02e8d416275b7006bbd584f9a2643ac8e051377/helpdesk/models.py#L589

**Reply by rodrigolfh** [reply]
It works! made a function at the top of models.py and called it right before the send()'s return statement.
It works like a charm, as it sends a whatsapp every time yo create/update/close a ticket.
